### PR TITLE
Add API for getting topology info as JSON

### PR DESCRIPTION
**Pull request by AnotherKamila**
This change is [Reviewable](https://reviewable.io/reviews/netsec-ethz/scionlab/300)

### CODE REVIEW

File: scionlab/urls.py [hunk 1]
```diff
@@ -79,6 +79,9 @@
     path('api/v2/host/<slug:uid>/deployed_config_version',
          PostHostDeployedConfigVersion.as_view(),
          name='api_post_deployed_version'),
+    path('api/v2/topology/topology.json',
```
Comment (FR4NK-W) [inline scionlab/urls.py:82]: When I see a file server with an extension, I usually expect that a static file is being served.
Also the file format might change, so the content type in the response should indicate what is being returned (JsonResponse does that), so remove the `.json` from the path.

**Review by FR4NK-W — CHANGES_REQUESTED** [review]

**Review by cmeury — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/netsec-ethz/scionlab/300)* status: 0 of 2 files reviewed, 3 unresolved discussions (waiting on @AnotherKamila)

___
*[scionlab/views/topology.py, line 127 at r1](https://reviewable.io/reviews/netsec-ethz/scionlab/300#-MG2H-h8E4u3T8YWkkvw:-MG2H-h8E4u3T8YWkkvx:b-2bfne3) ([raw file](https://github.com/netsec-ethz/scionlab/blob/b8e7903f7612e908dee78e9ca32290e64a85a8a4/scionlab/views/topology.py#L127)):*
> ```Python
> 
> @cache_page(1 * 60 * 60)
> @cache_control(public=True, max_age=24 * 60 * 60)
> ```

I am doubtful whether this should be cached at all.  If we need to change things in the coordinator and run Ansible with a dynamic inventory, the changes should be visible immediately, otherwise this is a hindrance to operations and likely requires cumbersome and error-prone manual changes in Ansible's cached inventory.

I'd suggest removing caching completely. What do you think?
___
*[scionlab/views/topology.py, line 137 at r1](https://reviewable.io/reviews/netsec-ethz/scionlab/300#-MG2TT4C36V2ng_ij1pD:-MG2TT4D1CdZunR6EHoX:b5yos1f) ([raw file](https://github.com/netsec-ethz/scionlab/blob/b8e7903f7612e908dee78e9ca32290e64a85a8a4/scionlab/views/topology.py#L137)):*
> ```Python
>         if service_type == 'CS': return s.port()+200
>         if service_type == 'BW': return None  # does not expose
>         if service_type == 'BR': return s.internal_port+400
> ```

How does this work? We distribute the toml files to the machines via the `scionlab-config` script, so I assume the coordinator sets the metrics ports. Couldn't we re-use the code that defines the ports here so it's less prone to fail upon future changes?

**Review by AnotherKamila — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/netsec-ethz/scionlab/300)* status: 0 of 2 files reviewed, 3 unresolved discussions (waiting on @cmeury and @FR4NK-W)

___
*[scionlab/urls.py, line 82 at r1](https://reviewable.io/reviews/netsec-ethz/scionlab/300#-MFpg3Qe-NejLfdxrw0w-r1-82:-MGmxRyA695lWMcDWnXu:b-896fix) ([raw file](https://github.com/netsec-ethz/scionlab/blob/b8e7903f7612e908dee78e9ca32290e64a85a8a4/scionlab/urls.py#L82)):*
<details><summary><i>Previously, FR4NK-W wrote…</i></summary><blockquote>

When I see a file served with an extension, I usually expect that a static file is being served.
Also the file format might change, so the content type in the response should indicate what is being returned (JsonResponse does that), so remove the `.json` from the path.
</blockquote></details>

Done.
___
*[scionlab/views/topology.py, line 127 at r1](https://reviewable.io/reviews/netsec-ethz/scionlab/300#-MG2H-h8E4u3T8YWkkvw:-MGmxXHQEIY0SOwUjAu7:bkh4rcs) ([raw file](https://github.com/netsec-ethz/scionlab/blob/b8e7903f7612e908dee78e9ca32290e64a85a8a4/scionlab/views/topology.py#L127)):*
<details><summary><i>Previously, cmeury (Cedric Meury) wrote…</i></summary><blockquote>

I am doubtful whether this should be cached at all.  If we need to change things in the coordinator and run Ansible with a dynamic inventory, the changes should be visible immediately, otherwise this is a hindrance to operations and likely requires cumbersome and error-prone manual changes in Ansible's cached inventory.

I'd suggest removing caching completely. What do you think?
</blockquote></details>

That's a good point. I copied that from the PNG generation above, but I guess that was silly. I don't want to completely remove caching because it still is a lot of DB calls, but I changed the time to 5s.

Arguably since our cron script is the only consumer of this API for now, I could completely remove caching, but it _is_ a lot of DB calls :D I'm really unsure though, so feel free to change my mind :D
___
*[scionlab/views/topology.py, line 137 at r1](https://reviewable.io/reviews/netsec-ethz/scionlab/300#-MG2TT4C36V2ng_ij1pD:-MGn0WfB-OdT9JmmaxHi:bl2v1pd) ([raw file](https://github.com/netsec-ethz/scionlab/blob/b8e7903f7612e908dee78e9ca32290e64a85a8a4/scionlab/views/topology.py#L137)):*
<details><summary><i>Previously, cmeury (Cedric Meury) wrote…</i></summary><blockquote>

How does this work? We distribute the toml files to the machines via the `scionlab-config` script, so I assume the coordinator sets the metrics ports. Couldn't we re-use the code that defines the ports here so it's less prone to fail upon future changes?
</blockquote></details>

That's a good point. I changed the magic constants into values from `scionlab.defines`, but in an ideal world we wouldn't even duplicate the logic for "this port is a costant and that one is an offset". Does anyone want to chime in on the "quickly vs properly" tradeoff?

**Review by cmeury — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/netsec-ethz/scionlab/300)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @AnotherKamila, @cmeury, and @FR4NK-W)

___
*[scionlab/views/topology.py, line 137 at r1](https://reviewable.io/reviews/netsec-ethz/scionlab/300#-MG2TT4C36V2ng_ij1pD:-MGnJHOT17T9rTPwWgyL:b-68rkni) ([raw file](https://github.com/netsec-ethz/scionlab/blob/b8e7903f7612e908dee78e9ca32290e64a85a8a4/scionlab/views/topology.py#L137)):*
<details><summary><i>Previously, AnotherKamila (Kamila Součková) wrote…</i></summary><blockquote>

That's a good point. I changed the magic constants into values from `scionlab.defines`, but in an ideal world we wouldn't even duplicate the logic for "this port is a costant and that one is an offset". Does anyone want to chime in on the "quickly vs properly" tradeoff?
</blockquote></details>

Using the same values from `defines` looks good to me.

**Review by cmeury — APPROVED** [review]

**Review by matzf — APPROVED** [review]
> Reviewed 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/netsec-ethz/scionlab/300)* status: all files reviewed, 1 unresolved discussion (waiting on @FR4NK-W)

**Pull request merged**